build/miengine.targets

```diff
@@ -7,7 +7,47 @@
 
   <PropertyGroup>
     <CoreCompileDependsOn>GenerateAssemblyInfoFile;$(CoreCompileDependsOn)</CoreCompileDependsOn>
+    <BuildDependsOn Condition="'$(IsCoreClr)' == 'false'">DetermineCompilerBinary;$(BuildDependsOn)</BuildDependsOn>
   </PropertyGroup>
+
+  <UsingTask TaskName="GetRuntime" 
+             TaskFactory="CodeTaskFactory" 
+             AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.v4.0.dll" >
+    <ParameterGroup>
+      <IsMonoRuntime ParameterType="System.Boolean" Output="true" />
+    </ParameterGroup>
+    <Task>
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[
+        IsMonoRuntime = System.Type.GetType("Mono.Runtime") != null;
+        ]]>
+      </Code>
+    </Task>
+  </UsingTask>
+
+  <Target Name="DetermineCompilerBinary">
+    <GetRuntime>
+      <Output PropertyName="IsMonoRuntime" TaskParameter="IsMonoRuntime" />
+    </GetRuntime>
+
+    <Exec Condition="'$(IsMonoRuntime)' == 'true'"
+          ConsoleToMsBuild="true"
+          Command="dirname `which mcs`">
+      <Output TaskParameter="ConsoleOutput" PropertyName="McsPath" />
+    </Exec>
+
+    <PropertyGroup Condition="'$(IsMonoRuntime)' == 'true'">
```

Comment:
This is just to enable building on Linux/OSX for a faster dev loop - official binaries would be built on Windows.